### PR TITLE
Improve error messages for invalid version ranges operators

### LINKED ISSUE #16066: [bug] Exact version range `[==...]` not working anymore with conan2
### Describe the bug

Hello,
I'm currently in the process of upgrading recipes from conan1 to conan 2 and I found out that `package/[==1.0.0]` is no longer working with conan2 (but I couldn't find anything about that in the doc) and conan doesn't throw an error like for `[~=1.0]` which is now explicitly invalid.

```
ERROR: Package 'package_name/[==1.0.0]@pypi/stable' not resolved: Version range '==1.0.0' from requirement 'package_name/[==1.0.0]@pypi/stable' required by ... could not be resolved.
```

It would be nice to either throw a proper error message ("this version range is no longer supported, please upgrade to ..." and update the doc) or fix the bug and re-add support for it.

### How to reproduce it

`conan install --requires "any_pkg/[==1.0.0]@user/channel"` vs `conan install --requires "any_pkg/1.0.0@user/channel"`

### PR DESCRIPTION
**Pull request by memsharded**
Changelog: Fix: Improve the definition of version ranges UX with better error message for invalid ``==, ~=, ^=`` operators.
Docs: Omit

Close https://github.com/conan-io/conan/issues/16066

### CODE REVIEW

File: conans/model/version_range.py [hunk 1]
```diff
@@ -88,6 +88,8 @@ def _parse_expression(expression):
             if expression[1] == "=":
                 operator += "="
                 index = 2
+        elif expression[1] == "=":
+            raise ConanException(f"Invalid version range operator '{operator}=' in {expression}")
```
Comment (araffin) [inline conans/model/version_range.py]:
```suggestion
            raise ConanException(f"Invalid version range operator '{operator}=' in {expression}, you should probably use {operator} instead.")
```
Maybe give a hint to the user on what is a correct range (the proposed error messages doesn't work in all cases, for instance for the `==` one)

Comment (araffin) [reply conans/model/version_range.py]: a better error message would probably be to distinguish between `~=`, `^=`, `==` and the rest.

File: conans/test/unittests/model/version/test_version_range.py [hunk 2]
```diff
@@ -5,6 +5,7 @@
 from conans.model.version_range import VersionRange
 
 values = [
+    ['=1.0.0',  [[['=', '1.0.0']]],                   ["1.0.0"],          ["0.1"]],
```
Comment (araffin) [inline conans/test/unittests/model/version/test_version_range.py:8]: so `[=1.0.0]` is supported?

Comment (memsharded) [reply conans/test/unittests/model/version/test_version_range.py:8]: Yes, I was a bit surprised too, seems more an unexpected result of the parsing, but I am concerned that removing now might break someone, so probably not worth it.

**Review by AbrilRBS — APPROVED** [review]

**Review by franramirez688 — APPROVED** [review]

**Pull request merged**